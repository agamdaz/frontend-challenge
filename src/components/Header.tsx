import React from "react";
import {media, style} from "typestyle";

const header = style({
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'space-between',
  padding: '8px',
  border: '2px solid #000',
})

const inputText = style({
  margin: '8px',
  padding: '8px',
  maxWidth: '100%',
  background: '#ececec',
  border: 'none',
  borderBottom: '2px solid #000',
  fontFamily: 'Courier Prime, monospace',
  fontWeight: 600,
  $nest: {
    '&:focus': {
      outlineColor: 'transparent',
      outlineStyle: 'none',
    },
  }
}, media({ maxWidth: 420 }, { flexGrow: 1 }))

const button = style({
  margin: '8px',
  padding: '8px',
  background: '#fff',
  border: '2px solid #000',
  fontFamily: 'Courier Prime, monospace',
  fontWeight: 600,
  cursor: 'pointer',
  $nest: {
    '&:focus': {
      outlineColor: 'transparent',
      outlineStyle: 'none',
    },
  }
}, media({ maxWidth: 420 }, { flexGrow: 1 }))

type HeaderProps = {
  inputPlaceholder: string,
  buttonLabel: string,
}

export const Header = (props: HeaderProps) => (
  <header data-testid="app-header" className={header}>
    <input className={inputText} type="text" placeholder={props.inputPlaceholder}/>
    <button className={button}>{props.buttonLabel}</button>
  </header>
);
