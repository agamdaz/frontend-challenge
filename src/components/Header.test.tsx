import React from 'react';
import { render, screen } from '@testing-library/react';
import { Header } from './Header'

test('Renders Header component with clear active', () => {
  render(<Header
    buttonLabel="AnyButtonLabel"
    inputPlaceholder="AnyPlaceholder"
    clearButtonActive={false}
    clearButtonClick={() => jest.isMockFunction}
    filterType={jest.isMockFunction}
  />);
  expect(screen.getByTestId('app-header')).toMatchSnapshot();
});

test('Renders Header component with clear disabled', () => {
  render(<Header
    buttonLabel="AnyButtonLabel"
    inputPlaceholder="AnyPlaceholder"
    clearButtonActive={true}
    clearButtonClick={() => jest.isMockFunction}
    filterType={jest.isMockFunction}
  />);
  expect(screen.getByTestId('app-header')).toMatchSnapshot();
});
