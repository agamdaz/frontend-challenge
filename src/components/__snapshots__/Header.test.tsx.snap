// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header component 1`] = `
<header
  class="fgayq2g"
  data-testid="app-header"
>
  <input
    class="f10tqc7z"
    placeholder="AnyPlaceholder"
    type="text"
  />
  <button
    class="f18k3v34"
  >
    AnyButtonLabel
  </button>
</header>
`;
